packages/telegram-cloud-storage-stamper/src/index.ts

```diff
@@ -0,0 +1,300 @@
+      );
+    }
+
+    return telegramStamper;
+  }
+
+  async stamp(payload: string) {
+    // check to see if we're in a telegram mini app context
+    this.checkTelegramContext();
+
+    // check to see that the stamper was initialized
+    if (!this.stamper) {
+      throw new TelegramCloudStorageStamperError(
+        "Cannot stamp with uninitialized telegram stamper, try running .create() or .setSigningKey()"
+      );
+    }
+
+    return await this.stamper.stamp(payload);
+  }
+
+  // set the API key that is used by the telegram cloud storage for signing requests
+  async setSigningKey(config?: TTelegramCloudStorageStamperConfig) {
+    if (!config) {
+      // try to get API key at default location and set the stamper to use that
+      // passing the DEFAULT_TURNKEY_CLOUD_STORAGE_KEY is not necessary since getAPIKey will use that by default
+      let { apiPublicKey, apiPrivateKey } = await this.getAPIKey(
+        DEFAULT_TURNKEY_CLOUD_STORAGE_KEY
+      );
+
+      if (apiPublicKey && apiPrivateKey) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey,
+          apiPrivateKey,
+        });
+        return true;
+      }
+    } else if (config.cloudStorageKey && !config.cloudStorageAPIKey) {
+      // try to get API key at specified location and set the stamper to use that
+      let { apiPublicKey, apiPrivateKey } = await this.getAPIKey(
+        config.cloudStorageKey
+      );
+
+      if (apiPublicKey && apiPrivateKey) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey,
+          apiPrivateKey,
+        });
+        return true;
+      }
+    } else if (!config.cloudStorageKey && config.cloudStorageAPIKey) {
+      // try to set API key at default location and set the stamper to use that
+      // passing the DEFAULT_TURNKEY_CLOUD_STORAGE_KEY is not necessary since insertAPIKey will use that by default
+      let stored = await this.insertAPIKey(
+        config.cloudStorageAPIKey.apiPublicKey,
+        config.cloudStorageAPIKey.apiPrivateKey,
+        DEFAULT_TURNKEY_CLOUD_STORAGE_KEY
+      );
+
+      if (stored === true) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey: config.cloudStorageAPIKey.apiPublicKey,
+          apiPrivateKey: config.cloudStorageAPIKey.apiPrivateKey,
+        });
+        return true;
+      }
+    } else if (config.cloudStorageKey && config.cloudStorageAPIKey) {
+      // try to set API key at specified location and set the stamper to use that
+      let stored = await this.insertAPIKey(
+        config.cloudStorageAPIKey.apiPublicKey,
+        config.cloudStorageAPIKey.apiPrivateKey,
+        config.cloudStorageKey
+      );
+
+      if (stored === true) {
+        this.stamper = new ApiKeyStamper({
+          apiPublicKey: config.cloudStorageAPIKey.apiPublicKey,
+          apiPrivateKey: config.cloudStorageAPIKey.apiPrivateKey,
+        });
+        return true;
+      }
+    }
+    return false;
```

Comment:
let's throw an exception here, guessing this is not supposed to happen!

Reply:
throwing